File: 13-roman-to-integer/roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        mp = {}
        sum = 0

        mp['I'] = 1
        mp['V'] = 5
        mp['X'] = 10
        mp['L'] = 50
        mp['C'] = 100
        mp['D'] = 500
        mp['M'] = 1000    

        for i in range(0,len(s),1):

            if(i+1 < len(s) and mp[s[i]] < mp[s[i+1]]):
                sum -= mp[s[i]]
            else:
                sum += mp[s[i]]
        
        return sum

        

File: 13-roman-to-integer/roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
       
        Map<Character,Integer> mp = new HashMap<>();
        mp.put('I',1);
        mp.put('V',5);
        mp.put('X',10);
        mp.put('L',50);
        mp.put('C',100);
        mp.put('D',500);
        mp.put('M',1000);

        int sum = 0;

        for(int i=0;i<s.length();i++)
        {
            if(i+1<s.length() && mp.get(s.charAt(i)) < mp.get(s.charAt(i+1)))
            {
                sum -= mp.get(s.charAt(i));
            }
            else
            {
                sum += mp.get(s.charAt(i));
            }

        }

        return sum;
        
    }
}